Calculator/main.js

```diff
@@ -0,0 +1,126 @@
+class Calculator {
+constructor(previousOperandTextElement, currentOperandTextElement) {
+    this.previousOperandTextElement = previousOperandTextElement
+    this.currentOperandTextElement = currentOperandTextElement
+    this.clear()
+}
+
+clear() {
+    this.currentOperand = ''
+    this.previousOperand = ''
+    this.operation = undefined
+}
+
+delete() {
+    this.currentOperand = this.currentOperand.toString().slice(0, -1)
+}
+
+appendNumber(number) {
+    if (number === '.' && this.currentOperand.includes('.')) return
+    this.currentOperand = this.currentOperand.toString() + number.toString()
+}
+
+chooseOperation(operation) {
+    if (this.currentOperand === '') return
```

Comment:
`if` leads to boolean conversion, which means that it converts value to true or false. Such values as null, undefined, NaN, "", 0 will give you false. (https://learn.javascript.ru/type-conversions#logicheskoe-preobrazovanie)
So you can check if (!this.currentOperand) return.
This way you'll avoid not only '', but also undefined, null and so on.

Reply:
fixed